src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown

```diff
@@ -0,0 +1,295 @@
+Pages](https://pages.github.com) for demo purposes.
+To add HTTPS to your server you'll need to get a TLS certificate and set
+it up. Be sure to check out [Security with HTTPS
+article](/web/fundamentals/security/)
+for best practices there.
+
+### User Gesture Required
+
+As a security feature, getting access to connected USB devices with
+`navigator.usb.requestDevice` must be called via a user gesture
+like a touch or mouse click.
+
+## Let's start coding
+
+The WebUSB API relies heavily on JavaScript
+[Promises](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+If you're not familiar with them, check out this great
+[Promises tutorial](http://www.html5rocks.com/en/tutorials/es6/promises/). One
+more thing, `() => {}` are simply ECMAScript 2015 [Arrow
+functions](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+-- they have a shorter syntax compared to function expressions and lexically
+bind the value of `this`.
+
+### Get Access to USB devices
+
+You can either prompt the user to select a single connected USB device using
+`navigator.usb.requestDevice` or call `navigator.usb.getDevices` to get a list
+of all connected USB devices the origin has access to.
+
+The `navigator.usb.requestDevice` function takes a mandatory JavaScript object
+that defines `filters`. These filters are used to match any USB device with the
+given vendor (`vendorId`) and optionally product (`productId`) identifiers. The
+`classCode`, `subclassCode` and `protocolCode` keys can also be defined there
+as well.
+
+<img style="width:723px; max-height:414px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/usb-device-chooser.png" alt="USB Device Chooser screenshot"/>
+
+For instance, here's how to get access to a connected Arduino device configured
+to allow the origin.
+
+{% highlight javascript %}
+navigator.usb.requestDevice({ filters: [{ vendorId: 0x2341 }] })
+.then(device => {
+  console.log(device.productName);      // "Arduino Micro"
+  console.log(device.manufacturerName); // "Arduino LLC"
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+Before you ask, I didn't magically come up with this `0x2341` hexadecimal
+number. I simply searched for the word "Arduino" in the official [List of USB
+ID's](http://www.linux-usb.org/usb.ids).
+
+The USB `device` returned in the fulfilled promise above has some basic, yet
+important information about the device such as the supported USB version,
+maximum packet size, vendor and product IDs, the number of possible
+configurations the device can have - basically all fields contained in the
+[device USB Descriptor](http://www.beyondlogic.org/usbnutshell/usb5.shtml#DeviceDescriptors)
+
+For info, if a USB device announces its [support for WebUSB](https://wicg.github.io/webusb/#webusb-platform-capability-descriptor)
+and as well as define a landing page URL, Chrome will show a persistent
```

Comment:
`For info, if a USB device announces its [support for WebUSB](https://wicg.github.io/webusb/#webusb-platform-capability-descriptor)
, as well as defining a landing page URL` I think?